test/verify/check-machines-dbus

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+import unittest
+
+from testlib import *
+import machineslib
+
+
+class TestMachinesDBus(machineslib.TestMachines):
+
+    def detectLibvirtDbus(self):
+        m = self.machine
+        namesList = m.execute("dbus-send --system --dest=org.freedesktop.DBus "
+                              "--type=method_call "
+                              "--print-reply /org/freedesktop/DBus "
+                              "org.freedesktop.DBus.ListActivatableNames")
+
+        if ('org.libvirt' not in namesList):
+            raise unittest.SkipTest("libvirt-dbus is not available");
+        m.execute("echo 'PROVIDER=libvirt-dbus' > /etc/environment")
+
+    def allowSystemDBusUnprivileged(self):
+        m = self.machine
+        libvirtConfig = '<?xml version="1.0"?> \
+        <!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN" \
+        "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd"> \
+        <busconfig> \
+        <policy user="admin"> \
+            <allow send_destination="org.libvirt"/> \
+        </policy> \
+        </busconfig>'
+        m.execute("echo '{0}' > {1}".format(libvirtConfig, "/etc/dbus-1/system.d/org.libvirt.conf"))
```

Comment:
This needs some explanation. admin can sudo and is a polkit admin too, so why doesn't libvirt-dbus'es polkit policy do the same as what `virsh -c qemu:///system` does? Or if it's really meant to be restricted to the root user (ugh, who does that), the tests should also reflect what happens in reality -- i. e. users wouldn't muck around with the policy file, but you would log in as a sudoer. And that's what the tests usually do. This looks like the code is missing a `{ superuser: "try" }` or even `"require"` somewhere?

Reply:
So, in order to have access as a normal user to system libvirt dbus the user needs either this file or he needs to be part of the libvirt group.

The admin user in the images we are using is not part of the libvirt group, so I decided to do the change as a part of the test suite only.

Note that for the same reason (admin user is not a part of the libvirt group) he doesn't have access to the system libvirt with virsh either.

See output below from the debian-testing machine:
```
root@unassigned-hostname:~# id admin 
uid=1000(admin) gid=1000(admin) groups=1000(admin),27(sudo)
root@unassigned-hostname:~# su - admin
admin@unassigned-hostname:~$ virsh -c qemu:///system
==== AUTHENTICATING FOR org.libvirt.unix.manage ===
System policy prevents management of local virtualized systems
Authenticating as: Administrator (admin)
Password: 

```

Reply:
So I can either leave it like this or change all images to add the admin user to libvirt group.

Reply:
No need to change the images -- just do the `usermod -a -G libvirt admin` call in the dbus test class'es `setUp()`. As long as you do this before `login_and_go()`, it will apply to the cockpit session.

Reply:
Done. Just a note that this bahavior is available since libvirt-dbus v1.2.0.
debian-testing has this version, but current fedora-testing image  not.
fedora-* images should be updated with newer libvirt-dbus (and libvirt for the bugfix that got backported) before the tests get enabled for fedora.